cmd/hubagent/main.go

```diff
@@ -42,6 +43,8 @@ var (
 		handleExitFunc()
 		os.Exit(1)
 	}
+
+	wg sync.WaitGroup
```

Comment:
declare the wg in the main func instead? otherwise, the whole main package can update the wg.